## Source attribution of Campylobacteriosis using Machine learning
Animal and environmental pathogens such as Campylobacter jejuni contaminate the food chain, making them leading causes of gastro-enteritis. We can use genetics to help attribute the source of a gastro-entritis outbreak in order to help policy makers to contain the spread by choosing the most appropriate control measures . The aim of this project is to develop genome-wide approaches to source attribution with a special focus on C. jejuni using the growing wealth of sequencing data combined with Machine Learning (ML), a toolbox still underused by evolutionary genomics.

In order to classify pathogen sequencing data taken from human clinical samples into their respective source reservoirs, multiple supervised and unsupervised ML techniques will be used on assembled genomes or DNA sequencing read data. A sequence feature matrix will be constructed using k-mer frequencies and additional basic sequence statistics like base frequency and GC-content. K-mer frequencies are generated by sliding a window of size k over the sequence and counting the occurrence of every unique subsequence of length k. K-mers have been shown to capture substitutions and indels (insertions/deletions) including whole-gene presence or absence in bacterial genomes and can be used to assign taxonomic labels to sequence samples. The proposed repertoire of classifiers utilised here comprises of Mixed linear models, Support Vector Machines, Decision Trees, Random Forests, Naive Bayesian Classifier, K-nearest Neighbours , Artificial Neural Networks, Bagging and Boosting 

The script ML_dimred.py throws a mixed bag of classifiers at the problem after initially performing a dimensionality reduction with a non-negative matrix factorisation. Accuracies of about 66% for source attribution can be achieved.
